### ISSUE TITLE
Standard PDF Form Fields

### ISSUE THREAD

**Issue by ivelin** [pdfme/pdfme#67]
First of all, congrats for the incredible work! Very impressive!

Noticed that Forms are saved in Templates which are in json format with binary payload and structured metadata. Has there been a discussion to support standard PDF Form fields for better interoperability with other PDF software?

For example pdf-lib has extensive API support for pdf form fields:
https://github.com/Hopding/pdf-lib#fill-form

Thank you!

Ivelin

**Comment by hand-dot** [comment]
Hi, @ivelin 

> Has there been a discussion to support standard PDF Form fields for better interoperability with other PDF software?
Not yet... But this idea is interesting!
Does it mean that the created PDF will be a PDF with values set in the PDF form?

**Comment by ivelin** [comment]
> Hi, @ivelin Thank you for the question.
> 
> > Has there been a discussion to support standard PDF Form fields for better interoperability with other PDF software?
> 
> Not yet... But this idea is interesting! Does it mean that the created PDF will be a PDF with values set in the PDF form?

Yes. The PDF Form fields can be empty with edit enabled or populated and read only after. See details in the pdf-lib docs:
https://pdf-lib.js.org/docs/api/classes/pdfform

**Comment by bkwdesign** [comment]
Yes, would be amazing to use your designer app, and when I change out the Document Template, once I upload it into your designer: it ought to auto-detected existing standard PDF form fields and surface the design information into your JSON format to allow for editing of the original PDF form

**Comment by psociety** [comment]
lol, that would be so cool, i didn't know pdf lib has those capabilities already

**Comment by faltenberg** [comment]
I fully agree with ivelin. I had such a case in my current project. We needed to generate a PDF with forms that the client could edit and we would process the edited PDF after and insert the data into the DB. I used jspdf as it was the only one I found with support for forms. A pain to work with though. More than 1000 lines for just 4 pages.

**Comment by hand-dot** [comment]
Version 3 has been feature released. It might be possible to implement by defining the new custom schemas feature.
https://pdfme.com/docs/custom-schemas

**Comment by jedimonkey** [comment]
Can confirm that this is possible, I've done a proof of concept for a project I'm working on. Will look at sharing the solution (or maybe a bonafide plugin!) if time allows me to. Essentially I just made a quick and nasty monkeypatched text plugin and supplied a new pdfRender function (essentially the couple of lines from the pdf lib link above) and worked a treat!

**Comment by hand-dot** [comment]
@jedimonkey Awesome! If it becomes possible, please share the code with us.

**Comment by hand-dot** [comment]
Closing as implemented for the original create standard PDF form fields request. Current @pdfme/generator exports generateForm, which creates AcroForm fields for editable text, checkbox, and radioGroup schemas. The remaining broader request around detecting/editing existing AcroForm fields is tracked separately in #1187.
